### DISCUSSION TITLE
ZeroDivisionError

### DISCUSSION THREAD

**Discussion by lea-danics** [nel-lab/mesmerize-core#288] [Q&A] [answered]
Dear All!
I just installed mescore and started with the notebook demo cnmfe.ipynb. I must admit, I'm a total beginner in programming in every possible aspects! :( I'm a wetlab researcher, but I would like try mesmerize to evaluate Ca imaging data.

By the following step I receive zero division error and i cannot overcome this issue:

"# input movie will be shown on left, filtered on right"
iw_gs = fpl.ImageWidget(
    data=[input_movie, input_movie.copy()],
    frame_apply=funcs,
    names=["raw", "filtered"],
    grid_plot_kwargs={"size": (1200, 600)},
    cmap="gnuplot2"
    

The detailed error message is the following:
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
Cell In[37], line 2
      1 # input movie will be shown on left, filtered on right
----> 2 iw_gs = fpl.ImageWidget(
      3     data=[input_movie, input_movie.copy()],
      4     frame_apply=funcs,
      5     names=["raw", "filtered"],
      6     grid_plot_kwargs={"size": (1200, 600)},
      7     cmap="gnuplot2"
      8 )

File [~\anaconda3\envs\mescore\lib\site-packages\fastplotlib\widgets\image.py:561](http://localhost:8888/~/anaconda3/envs/mescore/lib/site-packages/fastplotlib/widgets/image.py#line=560), in ImageWidget.__init__(self, data, dims_order, slider_dims, window_funcs, frame_apply, grid_shape, names, grid_plot_kwargs, histogram_widget, **kwargs)
    557 # update the default kwargs with any user-specified kwargs
    558 # user specified kwargs will overwrite the defaults
    559 grid_plot_kwargs_default.update(grid_plot_kwargs)
--> 561 self._gridplot: GridPlot = GridPlot(shape=grid_shape, **grid_plot_kwargs_default)
    563 for data_ix, (d, subplot) in enumerate(zip(self.data, self.gridplot)):
    564     if self._names is not None:

File [~\anaconda3\envs\mescore\lib\site-packages\fastplotlib\layouts\_gridplot.py:93](http://localhost:8888/~/anaconda3/envs/mescore/lib/site-packages/fastplotlib/layouts/_gridplot.py#line=92), in GridPlot.__init__(self, shape, cameras, controller_types, controller_ids, canvas, renderer, size, names)
     90 else:
     91     self.names = None
---> 93 canvas, renderer = make_canvas_and_renderer(canvas, renderer)
     95 if isinstance(cameras, str):
     96     # create the array representing the views for each subplot in the grid
     97     cameras = np.array([cameras] * self.shape[0] * self.shape[1]).reshape(
     98         self.shape
     99     )

File [~\anaconda3\envs\mescore\lib\site-packages\fastplotlib\layouts\_utils.py:89](http://localhost:8888/~/anaconda3/envs/mescore/lib/site-packages/fastplotlib/layouts/_utils.py#line=88), in make_canvas_and_renderer(canvas, renderer)
     83     raise ValueError(
     84         f"canvas option must either be a valid WgpuCanvas implementation, a pygfx Texture"
     85         f" or a str from the following options: {CANVAS_OPTIONS}"
     86     )
     88 if renderer is None:
---> 89     renderer = WgpuRenderer(canvas)
     91 return canvas, renderer

File [~\anaconda3\envs\mescore\lib\site-packages\pygfx\renderers\wgpu\_renderer.py:136](http://localhost:8888/~/anaconda3/envs/mescore/lib/site-packages/pygfx/renderers/wgpu/_renderer.py#line=135), in WgpuRenderer.__init__(self, target, pixel_ratio, pixel_filter, show_fps, blend_mode, sort_objects, enable_events, gamma_correction, *args, **kwargs)
    132     raise TypeError(
    133         f"Render target must be a Canvas or Texture, not a {target.__class__.__name__}"
    134     )
    135 self._target = target
--> 136 self.pixel_ratio = pixel_ratio
    137 self.pixel_filter = pixel_filter
    139 # Make sure we have a shared object (the first renderer creates the instance)

File [~\anaconda3\envs\mescore\lib\site-packages\pygfx\renderers\wgpu\_renderer.py:234](http://localhost:8888/~/anaconda3/envs/mescore/lib/site-packages/pygfx/renderers/wgpu/_renderer.py#line=233), in WgpuRenderer.pixel_ratio(self, value)
    232 target_lsize = self.logical_size
    233 # Determine target ratio
--> 234 target_ratio = target_psize[0] / target_lsize[0]
    235 # Use 2 on non-hidpi displays. On hidpi displays follow target.
    236 self._pixel_ratio = float(target_ratio) if target_ratio > 1 else 2.0

ZeroDivisionError: division by zero

Thanks in advance for your  help!

**Comment by kushalkolar** [comment, marked as answer]
Duplicate of https://github.com/nel-lab/mesmerize-core/issues/285
